packages/react-components/src/components/InputBoxComponent.tsx

```diff
@@ -201,3 +634,818 @@ export const InputBoxButton = (props: InputBoxButtonProps): JSX.Element => {
+        oldChangeEnd = oldTextLength - i - 1;
+        if (newText[newChangeEnd] !== oldText[oldChangeEnd]) {
+          // Change is found
+          break;
+        }
+      }
+      // make indexes exclusive
+      newChangeEnd += 1;
+      oldChangeEnd += 1;
+    }
+  } else {
+    //replacement
+    for (let i = 1; i < oldTextLength && oldTextLength - i >= changeStart; i++) {
+      newChangeEnd = newTextLength - i - 1;
+      oldChangeEnd = oldTextLength - i - 1;
+
+      if (newText[newChangeEnd] !== oldText[oldChangeEnd]) {
+        // Change is found
+        break;
+      }
+    }
+    // make indexes exclusive if they aren't equal to the length of the string
+    if (newChangeEnd !== newText.length) {
+      newChangeEnd += 1;
+    }
+    if (oldChangeEnd !== oldText.length) {
+      oldChangeEnd += 1;
+    }
+  }
+  return { changeStart, oldChangeEnd, newChangeEnd };
+};
+
+/* @conditional-compile-remove(mention) */
+const htmlStringForMentionSuggestion = (suggestion: Mention, localeStrings: ComponentStrings): string => {
+  const idHTML = ' id ="' + suggestion.id + '"';
+  const displayTextHTML = ' displayText ="' + suggestion.displayText + '"';
+  const displayText = getDisplayNameForMentionSuggestion(suggestion, localeStrings);
+  return '<msft-mention' + idHTML + displayTextHTML + '>' + displayText + '</msft-mention>';
+};
+
+/* @conditional-compile-remove(mention) */
+const getDisplayNameForMentionSuggestion = (suggestion: Mention, localeStrings: ComponentStrings): string => {
+  const displayNamePlaceholder = localeStrings.participantItem.displayNamePlaceholder;
+  return suggestion.displayText !== '' ? suggestion.displayText : displayNamePlaceholder ?? '';
+};
+
+/* @conditional-compile-remove(mention) */
+type TagData = {
+  tagType: string; // The type of tag (e.g. msft-mention)
+  openTagIdx: number; // Start of the tag relative to the parent content
+  openTagBody: string; // Complete open tag body
+  content?: string; // All content between the open and close tags
+  closeTagIdx?: number; // Start of the close tag relative to the parent content
+  subTags?: TagData[]; // Any child tags
+  plainTextBeginIndex?: number; // Absolute index of the open tag start should be in plain text
+  plainTextEndIndex?: number; // Absolute index of the close tag start should be in plain text
+};
+
+/* @conditional-compile-remove(mention) */
+type HtmlTagType = 'open' | 'close' | 'self-closing';
+/* @conditional-compile-remove(mention) */
+type HtmlTag = {
+  content: string;
+  startIdx: number;
+  type: HtmlTagType;
+};
+
+/* @conditional-compile-remove(mention) */
+/**
+ * Parse the text and return the tags and the plain text in one go
+ * @param text - The text to parse for HTML tags
+ * @param trigger The trigger to show for the msft-mention tag in plain text
+ *
+ * @returns An array of tags and the plain text representation
+ */
+const textToTagParser = (text: string, trigger: string): [TagData[], string] => {
```

Comment:
Not necessary in this PR, I believe we would need Unit tests for all the html parsing functions just to ensure everything works always right

Reply:
There is a task for this in the backlog; 100% agree.